File: 3301.php
<?php
  function getSobrinhoMeio($h, $z, $l) {
    if (($h > $z && $h < $l) || ($h < $z && $h > $l)) {
      return "huguinho" . PHP_EOL;
    }

    if (($z > $h && $z < $l) || ($z < $h && $z > $l)) {
      return "zezinho" . PHP_EOL;
    }
    
    return "luisinho" . PHP_EOL;
  }

  function main() {
    $entrada = trim(fgets(STDIN));
    
    list($h, $z, $l) = explode(" ", $entrada);

    echo(getSobrinhoMeio($h, $z, $l));
  }

  main();
?>

File: 3301.go
package main

import (
  "fmt"
)

func getSobrinhoMeio(h int, z int, l int) string {
  if (h > z && h < l) || (h < z && h > l) {
    return "huguinho"
  }

  if (z > h && z < l) || (z < h && z > l) {
    return "zezinho"
  }

  return "luisinho"
}

func main() {
  var h, z, l int
  fmt.Scanf("%d %d %d", &h, &z, &l)
  
  fmt.Printf("%s\n", getSobrinhoMeio(h, z, l))
}